packages/jest-core/src/__tests__/SearchSource.test.ts

```diff
@@ -206,6 +206,34 @@ describe('SearchSource', () => {
       });
     });
 
+    it('finds tests matching a JS with overriding glob patterns', () => {
+      const {options: config} = normalize(
+        {
+          moduleFileExtensions: ['js', 'jsx'],
+          name,
+          rootDir,
+          testMatch: [
+            '**/*.js?(x)',
+            '!**/test.js?(x)',
+            '**/test.js',
+            '!**/test.js',
+          ],
+          testRegex: '',
+        },
+        {} as Config.Argv,
+      );
+
+      return findMatchingTests(config).then(data => {
```

Comment:
Async await? Or, I guess this matches the style of the other tests?

Reply:
Yep, mostly copypasta here